### PR TITLE
Allows coloring objects with spray cans

### PR DESCRIPTION
**Pull request by Mickyan**
:cl: Mickyan
add: Most objects can now be colored using a spray can
spellcheck: Added visible message to spraying objects and windows
/:cl:

I noticed using crayon powder you can paint literally anything and even clothes show up as you'd expect when you wear them, but since you get a very limited range of colors this seemed like a wasted opportunity

paint

### CODE REVIEW

**Comment by imsxz** [comment]
how will this interact with the item(s) that use the spraypaint component? HECK suit and helmet are the only ones to come to mind, but the questions there.

**Comment by Mickyan** [comment]
component takes precedence, so no changes

**Comment by Angustmeta** [comment]
Just sure everything can be cleaned with space cleaner or soap. I remember space cleaner not working on tiles colored by colorful reagent at some point.

**Comment by DaxDupont** [comment]
red vehicles move faster so this is a buff

**Comment by ATH1909** [comment]
> red vehicles move faster so this is a buff

I actually want this to become a legitimate thing (although the speed buff should be very small).

**Comment by DaxDupont** [comment]
It is a thing, for scooters at least

**Comment by necromanceranne** [comment]
It sorta is in the form of red potions. They nake slow things move faster.

**Comment by anconfuzedrock** [comment]
>get blue military jacket
>get white balaclava
>wear medhud (sunglasses) under balaclava
>SANS!!!!!!!$!!!))!))!5/8:& mrkesc

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -651,10 +651,10 @@
 
 		return
 
-	if(istype(target, /obj/structure/window))
+	if(isobj(target))
 		if(actually_paints)
 			target.add_atom_colour(paint_color, WASHABLE_COLOUR_PRIORITY)
-			if(color_hex2num(paint_color) < 255)
+			if(color_hex2num(paint_color) < 255 && istype(target, /obj/structure/window))
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/crayons.dm]: what's the double check for?

File: code/game/objects/items/crayons.dm [hunk 2]
```diff
@@ -665,6 +665,7 @@
 
 		if(pre_noise || post_noise)
 			playsound(user.loc, 'sound/effects/spray.ogg', 5, 1, 5)
+		user.visible_message("[user] coats [target] with spray paint!", "<span class='notice'>You coat [target] with spray paint.</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/crayons.dm]: Your

**Comment by ExcessiveUseOfCobblestone** [comment]
help I need coffee

**Pull request merged**